File: content/search/source/includes/view/code-snippets/nodejs/filter-documents-query.js
import { MongoClient } from "mongodb";

// Connect to your Atlas deployment
const uri = "<connectionString>";
const client = new MongoClient(uri);

async function run() {
  try {
    const database = client.db("sample_mflix");
    const collection = database.collection("movies_ReleasedAfter2000");

    // Run the aggregation query on the View
    const pipeline = [
      {
        $search: {
          index: "releasedAfter2000Index",
          text: {
            path: "title",
            query: "foo"
          },
          sort: {
            released: 1
          }
        }
      }
    ];

    const results = await collection.aggregate(pipeline).toArray();
    console.log(results);
  } finally {
    await client.close();
  }
}
run().catch(console.dir);

File: content/search/source/includes/view/code-snippets/csharp/FilterDocumentsQuery.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Connect to your Atlas deployment
var uri = "<connectionString>";
var client = new MongoClient(uri);
var database = client.GetDatabase("sample_mflix");
var collection = database.GetCollection<BsonDocument>(
    "movies_ReleasedAfter2000");

// Run the aggregation query on the View
var pipeline = new[]
{
    new BsonDocument("$search",
        new BsonDocument("index", "releasedAfter2000Index")
            .Add("text",
                new BsonDocument("path", "title")
                    .Add("query", "foo"))
            .Add("sort",
                new BsonDocument("released", 1)))
};

var results = collection.Aggregate<BsonDocument>(pipeline);
foreach (var doc in results.ToList())
{
    Console.WriteLine(doc);
}
